src/components/interaction-designer/clipboard/prompt-kinds/Message.vue

```diff
@@ -17,58 +17,62 @@
     </div>
 </template>
 
-<script>
-import { IContext } from '@floip/flow-runner'
-import { mapActions, mapGetters } from 'vuex'
+<script lang="ts">
 import BlockActionButtons from '../shared/BlockActionButtons.vue'
+import Component, { mixins } from 'vue-class-component';
+import { Prop } from 'vue-property-decorator';
+import { namespace } from 'vuex-class';
+import { IPrompt } from '@floip/flow-runner';
+import Lang from '@/lib/filters/lang';
 
-export default {
+const clipboardVuexNamespace = namespace('clipboard')
+
+@Component({
   components: {
-    BlockActionButtons,
-  },
-  props: {
-    // TODO: use class inheritance to avoid repeating code when we transform this into vue-class` based component
-    context: IContext,
-    index: Number,
-    isComplete: Boolean,
-    goNext: Function,
-    onEditComplete: Function,
-  },
-  data() {
-    return {
-      isBlockInteraction: false,
-    }
+    BlockActionButtons
   },
-  computed: {
-    ...mapGetters('clipboard', ['isBlockFocused', 'getBlockPrompt']),
-    isFocused() {
-      return this.isBlockFocused(this.index)
-    },
-    prompt() {
-      return this.getBlockPrompt(this.index)
-    },
-  },
-  methods: {
-    ...mapActions('clipboard', ['setIsFocused', 'setLastBlockUnEditable', 'setLastBlockEditable']),
-    async submitAnswer() {
-      if (this.isBlockInteraction) {
-        await this.onEditComplete(this.index)
-        this.isBlockInteraction = false
-      }
-      this.prompt.value = null
-      this.setIsFocused({ index: this.index, value: false })
-      this.goNext()
-    },
-    editBlock() {
-      this.setIsFocused({ index: this.index, value: true })
-      this.setLastBlockUnEditable()
-      this.isBlockInteraction = true
-    },
-    onCancel() {
-      this.setLastBlockEditable()
-      this.setIsFocused({ index: this.index, value: false })
+})
+export default class Message extends mixins(Lang) {
```

Comment:
@poojahpatil90  , I'd like to leave a note for knowledge sharing, and in case we want some improvement in the future:

There are several repetitions in `prompt-kinds/....vue` components, as we're using vue-class based component, we can now use inheritance to reduce repetitions & enhance maintainability like:
- SelectMany inherits from SelectOne: https://github.com/FLOIP/flow-builder/blob/master/src/components/interaction-designer/block-types/MobilePrimitives_SelectManyResponseBlock.vue#L96
- Stories inherits from `BaseMountedVueClass` : https://github.com/FLOIP/flow-builder/blob/master/stories/MessageBlock.stories.ts#L37

Reply:
Done!